build_all/mac/release.sh

```diff
@@ -45,96 +51,138 @@ process_args()
+
+    # Set the default cores
+    CORES=$(sysctl -n hw.ncpu)
+
+    make --jobs=$CORES
+    ctest -j $CORES -C "Debug" --output-on-failure
+
+    popd
+}
 
-make --jobs=$CORES
-ctest -j $CORES -C "Debug" --output-on-failure
+copy_binaries()
+{
+    [ $? -eq 0 ] || exit $?
+    cd $build_root
+
+    echo copy iothub_client library to samples folder
+    cp $build_folder/python/src/iothub_client.dylib ./device/samples/iothub_client.so
+    echo copy iothub_client_mock library to tests folder
+    cp $build_folder/python/test/iothub_client_mock.dylib ./device/tests/iothub_client_mock.so
+    cp $build_folder/python/src/iothub_client.dylib ./device/tests/iothub_client.so
+    cp $build_folder/python_service_client/src/iothub_service_client.dylib ./device/tests/iothub_service_client.so
+
+    echo copy iothub_service_client library to samples folder
+    cp $build_folder/python_service_client/src/iothub_service_client.dylib ./service/samples/iothub_service_client.so
+    echo copy iothub_service_client_mock library to tests folder
+    cp $build_folder/python_service_client/tests/iothub_service_client_mock.dylib ./service/tests/iothub_service_client_mock.so
+    cp $build_folder/python_service_client/src/iothub_service_client.dylib ./service/tests/iothub_service_client.so
+    cp $build_folder/python/src/iothub_client.dylib ./service/tests/iothub_client.so
+}
 
-popd
-########################################################
+run_unit_tests()
+{
+    cd $build_root/device/tests/
+    echo "python${PYTHON_VERSION}" iothub_client_ut.py
+    "python${PYTHON_VERSION}" iothub_client_ut.py
+    [ $? -eq 0 ] || exit $?
+    echo "python${PYTHON_VERSION}" iothub_client_map_test.py
+    "python${PYTHON_VERSION}" iothub_client_map_test.py
+    [ $? -eq 0 ] || exit $?
+    cd $build_root
+
+    cd $build_root/service/tests/
+    echo "python${PYTHON_VERSION}" iothub_service_client_ut.py
+    "python${PYTHON_VERSION}" iothub_service_client_ut.py
+    [ $? -eq 0 ] || exit $?
+    echo "python${PYTHON_VERSION}" iothub_service_client_map_test.py
+    "python${PYTHON_VERSION}" iothub_service_client_map_test.py
+    [ $? -eq 0 ] || exit $?
+    cd $build_root
+}
 
+create_packages()
+{
+    cd ./build_all/mac/release_device_client
+    cp $build_folder/python/src/iothub_client.dylib iothub_client/iothub_client.so
+    "python${PYTHON_VERSION}" setup.py bdist_wheel --plat-name $PLAT_ARCH
+    cd $build_root
+
+    cd ./build_all/mac/release_service_client
+    cp $build_folder/python_service_client/src/iothub_service_client.dylib iothub_service_client/iothub_service_client.so
+    "python${PYTHON_VERSION}" setup.py bdist_wheel --plat-name $PLAT_ARCH
+    cd $build_root
+}
 
-[ $? -eq 0 ] || exit $?
-cd $build_root
+upload_packages()
```

Comment:
same here, needs error checking

Reply:
When we manually run the release script we always have to check the output for errors. 
The upload will output an error if any. We don't want tot go to the 3rd party toll's output parsing business.